rust/src/utils/datalink.rs

```diff
@@ -0,0 +1,77 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Jeff Lucovsky <jlucovsky@oisf.net>
+
+use std::collections::HashMap;
+use std::ffi::{CStr, CString};
+use std::os::raw::c_char;
+use std::ptr;
+
+fn get_datalink_value(map: &HashMap<i32, CString>, key: i32) -> Option<&CString> {
+    map.get(&key)
+}
+
+fn add_datalink_value(map: &mut HashMap<i32, CString>, key: i32, value: &CStr) {
+    if let Ok(str_value) = value.to_str() {
+        if let Ok(cstring_value) = CString::new(str_value) {
+            map.insert(key, cstring_value);
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn SCDatalinkInit() -> *mut HashMap<i32, CString> {
+    let map: HashMap<i32, CString> = HashMap::new();
```

Comment:
Q: How many datalink values can we expect to insert in the HashMap?
Or rather, what encouraged the use of a HashMap here?

Reply:
I chose a hash map to store the mappings between linktype values and the name as the link type values are established values that are non-sequential.  With the small number of entries (12-15), access should be o(1).

Reply:
But the key should rather be u8 or u16 than `i32`, right ?

Reply:
Datalink is i32 I believe, so it makes sense to use it here.

Reply:
actually u16 seems to be what is suggested here https://datatracker.ietf.org/doc/draft-ietf-opsawg-pcaplinktype/

however I think the point is really minor, and at this point I feel handling the int to u16 isn't really worth it right now